components/NewResourceForm.js

```diff
@@ -0,0 +1,82 @@
+import { useOperationMethod } from "react-openapi-client";
+import React, { useState, useCallback } from "react";
+import { TextField, Button, Box, CircularProgress, Alert } from "@mui/material";
+
+function NewResourceForm({ learningUnitId, handleClose }) {
+  const [createResource, { loading, error, data }] =
+    useOperationMethod("createResource");
+  const [formInput, setContent] = useState({ name: "", url: "" });
+
+  const handleChange = useCallback((event) => {
+    setContent((prevFormInput) => {
+      return {
+        ...prevFormInput,
+        [event.target.name]: event.target.value,
+      };
+    });
+  }, []);
+
+  const handleSubmit = () => {
+    createResource(learningUnitId, {
+      name: formInput.name,
+      url: formInput.url,
+    });
+  };
+
+  if (error) {
+    return (
+      <>
+        <Alert severity="error">
+          Error {error.response.data.code}: {error.response.data.message}
+        </Alert>
+        <Button onClick={handleClose}>Close</Button>
+      </>
+    );
+  }
+  if (loading) {
+    return (
+      <Box sx={{ display: "flex" }}>
+        <CircularProgress />
+      </Box>
+    );
+  }
+  if (data) {
+    return (
+      <>
+        <Alert severity="success">Resource {data.name} created!</Alert>
+        <Button onClick={handleClose}>Close</Button>
+      </>
+    );
+  }
+
+  return (
+    <>
+      <TextField
+        required
+        fullWidth
+        type="text"
+        id="name"
+        name="name"
+        label="Name your resource"
+        onChange={handleChange}
+        value={formInput.name}
+      />
+      <br />
```

Comment:
I think that some extra space between the boxes would be nice: 
<img width="493" alt="image" src="https://user-images.githubusercontent.com/73831187/193712586-f5298bb2-52ff-45cc-8816-e70b3912582d.png">
Maybe put some margin instead of just using a br

Reply:
Done